Tests/BinaryCompatibility.lean

```diff
@@ -0,0 +1,207 @@
+  test "This is a canary test. Chirp chirp." true
+
+def testCrow : TestSeq :=
+  -- test "This is a crow test. Caw caw." false
+  test
+    "You need to have `wasm-sandbox` binary in your current work directory. Please run `wget https://github.com/cognivore/wasm-sandbox/releases/download/v1/wasm-sandbox && chmod +x ./wasm-sandbox`."
+    false
+
+-- A very bad hasing function. It adds up the character codes of each of the string's characters, and then appends 'L' followed by the number of characters in the string to reduce the chance of collisions.
+def myHash (s : String) : String :=
+  let n := s.foldl (fun acc c => acc + c.toNat) 0
+  s!"{n}L{s.length}"
+
+def fname (s : String) : String :=
+  "./wast-dump-" ++ myHash s ++ ".bytes"
+
+def uWasmMods := [
+  "(module
+        (func (export \"main\")
+            (result i32)
+
+            (i32.add
+                (i32.const 1499550000)
+                (i32.add (i32.const 9000) (i32.const 17))
+            )
+        )
+    )",
+    "(module
+        (func (export \"two_ints\")
+            (result i32) (result i32)
+            (i32.add
+                (i32.const 1499550000)
+                (i32.add (i32.const 9000) (i32.const 17))
+            )
+            (i32.add (i32.const -1) (i32.const 1))
+        )
+    )",
+    "(module
+        (func (param $x_one i32) (param $three i32) (param $y_one i32) (result i32) (i32.add (i32.const 40) (i32.const 2)))
+        (func (param $x_two f32) (param f32) (param f32) (result i32) (i32.add (i32.const 12) (i32.const 30)))
+    )",
+    "(module
+      (func (param $x i32) (param i32) (result i32) (i32.add (i32.const 40) (i32.const 2)))
+    )",
+    "(module
+        ( func (param $x i32) (param i32) )
+      )",
+    "(module
+        (func (param $x i32))
+      )",
+    "(module
+        ( func )
+      )",
+    "(module
+      (func $main (export \"main\")
+        (param $x i32)
+        (param i32)
+        (result i32 i32)
+
+        (block (result i32) (i32.const 3) (i32.add (br 0) (i32.const 9)))
+        (i32.add
+          (i32.const 1499550000)
+          (i32.add (i32.const 9000) (i32.const 17))
+        )
+
+      )
+    )",
+    "(module
+      (func $main (export \"main\")
+        (param $x i32)
+        (param i32)
+        (result i32 i32)
+
+        (block (result i32) (i32.const 3) (i32.add (br 0) (i32.const 9)))
+        (i32.add
+          (i32.const 1499550000)
+          (i32.add (i32.const 17) (i32.const 9000))
+        )
+
+      )
+    )"
+]
+
+-- TODO: Prove termination because lengths decrease 1 at a time.
+-- Byte by byte, compare the two variables, reporting errors.
+partial def loop (rs : ByteArray) (os : ByteArray) : Bool :=
+  dbg_trace s!"loop:\n{rs}\n{os}\n***********************************"
+  let il := os.data.toList
+  let jl := rs.data.toList
+  -- Let's see if the lengths aren't the same. If they aren't, we're done.
+  match il.length != jl.length with
+  | true => false
+  | false => Id.run $ do
```

Comment:
`if`?

Reply:
I'll explain. I wanted to prove that one of the branches don't have to be considered here, I think, but failed and then I was too tired to rewrite to if.